### PR TITLE
Fix FieldTimeSeries interpolation for single column grids with x, y locations

### PR DESCRIPTION
**Pull request by glwagner**
This is needed for interpolating to "located" single column grids, which we use in `ClimaOcean` for examples.

### CODE REVIEW

File: src/Grids/grid_utils.jl [hunk 1]
```diff
@@ -175,7 +175,7 @@ constant grid spacing `Δ`, and interior extent `L`.
 
 # Grid domains
 @inline domain(topo, N, ξ) = CUDA.@allowscalar ξ[1], ξ[N+1]
-@inline domain(::Flat, N, ξ::AbstractArray) = CUDA.@allowscalar ξ[1], ξ[1]
+@inline domain(::Flat, N, ξ::AbstractArray) = ξ[1]
```
Comment (navidcy) [inline src/Grids/grid_utils.jl:178]: wouldn't that fail without the `@allowscalar`?

Comment (glwagner) [reply src/Grids/grid_utils.jl:178]: A one-element range is used to store the coordinate when we have a "located" `Flat` direction. Array / CuArray is only used for stretched coordinates. I can think of a use case but I'm inclined to say that we should cross that bridge when we get there...

Comment (navidcy) [reply src/Grids/grid_utils.jl:178]: ok!

**Comment by navidcy** [comment]
Is this ready to merge if tests pass?

**Comment by navidcy** [comment]
I'm suggesting to wait to include this PR in the v0.91.13 release

**Review by navidcy — APPROVED** [review]

**Pull request merged**